lib/validators/specResolver.ts

```diff
@@ -922,6 +922,8 @@ export class SpecResolver {
     // (constant) on property marked as discriminator
     const definition = definitions[name]
     if (definition && definition.properties) {
+      // all derived types should have `"type": "object"`.
+      definition.type = "object"
```

Comment:
can u please point to the documentation specifying this for future reference .
Also what happens if the definition is actually not an object ? are we going to surface that as an issue in the semantic validator ?

Reply:
We use z-schema for validation, if there is no `type`, z-schema assumes that it could be anything (ex. `string`, `number`, etc). In this case, almost any value (except object) will pass validation. Which's definitely is not what we want.